R/safe_get.R

```diff
@@ -0,0 +1,18 @@
+# safe version of GET, reads status code before continuing.
+safe_get <- function(feed, user = NULL, config = list()) {
+  safeget <- safely(GET)
+  req <- safeget(feed, user, config)
+
+  if (!is.null(req$error)) {
+    msg <- paste0("Attempt to fetch feed resulted in an error: ", req$error)
+    stop(msg)
+  }
+  status <- req$result$status_code
+  if (status != 200L) {
+    stop("Attempt to get feed was unsuccessful (non-200 response). Feed may not be available.")
```

Comment:
does `http::GET` handle 303 response automatically? that'd be a case I'd worry about with feeds by doing this.

Reply:
Yes, it seems to: 

```r
> httr::GET("https://httpstat.us/303")
Response [https://httpstat.us/]
  Date: 2020-02-24 19:57
  Status: 200
  Content-Type: text/html; charset=utf-8
  Size: 8.55 kB
<!DOCTYPE html>
<html>
...
```